### ISSUE TITLE
Rewrite in Java, drop Scala dependencies

### ISSUE THREAD

**Issue by almson** [mbknor/mbknor-jackson-jsonSchema#160]
This thing being written in Scala is a problem:

  - a bunch of extra dependencies
  - difficulty in making contributions

This library would be better if it was written in Java. To that end, I've went ahead and made a conversion.

**Comment by fdutton** [comment]
I agree that this project should be rewritten in Java. I also encountered difficulties with its use of Scala. Mainly with the conflicts created with other libraries that also use Scala but use different versions.

I think you are unlikely to succeed in this request given that you are asking this project to not only switch languages but also adopt Java 17. There are many clients of this project that still use Java 8. Even if you were successful in arguing that Java 8 has reached its end-of-life, Java 11 still has two years before it reaches that state. At a minimum, you would have to support Java 11. To make it a drop-in replacement, you have to support Java 8.

**Comment by almson** [comment]
@fdutton Java version is not a big deal. I wanted to see if there's interest. If so, feel free to participate in my repo.
